File: WoofAndroid/domain/model/src/main/kotlin/com/perrystreet/woof/models/dog/DogsFeed.kt
package com.perrystreet.woof.models.dog

data class DogsFeed(
    val dogs: List<Dog>,
    val total: Int,
) {
    val hasMore: Boolean
        get() = dogs.size < total

    companion object {
        val Empty = DogsFeed(dogs = emptyList(), total = 0)
    }
}

File: WoofIOS/SwiftPackages/Domain/Model/Sources/Models/Dog/DogsFeed.swift
import Foundation

public struct DogsFeed: Hashable {
    public let dogs: [Dog]
    public let total: Int

    public var hasMore: Bool {
        dogs.count < total
    }

    public static let empty = DogsFeed(dogs: [], total: 0)

    public init(dogs: [Dog], total: Int) {
        self.dogs = dogs
        self.total = total
    }
}
